lib/Dialect/FIRRTL/FIRRTLAnnotationHelper.cpp

```diff
@@ -341,17 +341,82 @@ InstanceOp firrtl::addPortsToModule(
     instancePathcache.replaceInstance(useInst, clonedInst);
     if (targetCaches)
       targetCaches->replaceOp(useInst, clonedInst);
-    if (instancePathToUpdate) {
-      auto *iter = llvm::find(instancePathToUpdate->instances, useInst);
-      if (iter != instancePathToUpdate->instances.end())
-        *iter = clonedInst;
-    }
     useInst->replaceAllUsesWith(clonedInst.getResults().drop_back());
     useInst->erase();
   }
   return clonedInstOnPath;
 }
 
+Value firrtl::borePortsOnPath(
+    SmallVector<InstanceOp> &instancePath, FModuleOp lcaModule, Value fromVal,
+    StringRef newNameHint, InstancePathCache &instancePathcache,
+    llvm::function_ref<ModuleNamespace &(FModuleLike)> getNamespace,
+    CircuitTargetCache *targetCachesInstancePathCache) {
+  // Create connections of the `fromVal` by adding ports through all the
+  // instances of `instancePath`. `instancePath` specifies a path from a source
+  // module through the lcaModule till a target module. source module is the
+  // parent module of `fromval` and target module is the referenced target
+  // module of the last instance in the instancePath.
```

Comment:
I really like the approach of having a single path of instances that goes up, through the LCA, and down the hierarchy. That removes a lot of ambiguity and forces users of `borePortsOnPath` to be very explicit.